package.json

```
@@ -18,12 +18,22 @@
     "@typescript-eslint/eslint-plugin": "^4.15.0",
     "@typescript-eslint/parser": "^4.15.0",
     "eslint": "^7.23.0",
+    "husky": "^7.0.1",
+    "lint-staged": "^11.1.1",
+    "prettier": "^2.3.2",
     "typescript": "^4.2.3"
   },
   "scripts": {
     "build": "tsc",
     "lint": "eslint index.ts",
-    "prepare": "rm -rf dist && npm run build"
+    "prepare": "rm -rf dist && npm run build",
+    "precommit": "lint-staged"
```

Husky v6以降の場合下記のような手順で .husky/pre-commit にコミットするのがよさそうに思います。ご検討ください。

```suggestion
    "prepare": "husky install && rm -rf dist && npm run build"
```

そして

```sh
npm run prepare
npx husky add .husky/pre-commit lint-staged
```

参考文献: https://github.com/typicode/husky#usage